src/api/fungibleAsset.ts

```diff
@@ -42,10 +43,31 @@ export class FungibleAsset {
       pagination?: PaginationArgs;
       where?: FungibleAssetMetadataBoolExp;
     };
-  }): Promise<GetFungibleAssetMetadataResponse> {
+  }): Promise<GetFungibleAssetMetadatasResponse> {
     return getFungibleAssetMetadata({ aptosConfig: this.config, ...args });
   }
 
+  /**
+   * Queries the current specific fungible asset metadata
+   *
+   * This query returns the fungible asset metadata for a specific fungible asset.
+   *
+   * @param assetType The asset type of the fungible asset
+   * @returns getFungibleAssetMetadata A list of fungible asset metadata
+   */
+  async getFungibleAssetMetadataByAssetType(assetType: string): Promise<GetFungibleAssetMetadataResponse> {
+    const data = await getFungibleAssetMetadata({
+      aptosConfig: this.config,
+      options: {
+        where: {
+          asset_type: { _eq: assetType },
+        },
+      },
+    });
+
+    return data[0];
```

Comment:
is there more than one?  Then let's return all and leave to the user to handle it, or provide a assertion or error that handles when there's more than one.